script.js

```diff
@@ -1,4 +1,493 @@
+    playerProfiles[currentPlayer].playerAceCount += checkAce(card);
+
+    playerProfiles[currentPlayer].playerHand +=
+      "<br>" + card.value + " of " + card.suits;
+
+    while (
+      playerProfiles[currentPlayer].playerHandValue > 21 &&
+      playerProfiles[currentPlayer].playerAceCount > 0
+    ) {
+      playerProfiles[currentPlayer].playerHandValue -= 10;
+      playerProfiles[currentPlayer].playerAceCount -= 1;
+    }
+
+    if (playerProfiles[currentPlayer].playerHandValue <= 21) {
+      gameMode = GAME_MODE_HIT_OR_STAND;
+      var playerHandmessage = createPlayerHandmessage();
+      return (
+        playerHandmessage +
+        "<br><br> Player " +
+        playerProfiles[currentPlayer].id +
+        ", it's still your turn, continue to hit or stand? "
+      );
+    } else if (
+      playerProfiles[currentPlayer].playerHandValue > 21 &&
+      currentPlayer + 1 < numPlayers
+    ) {
+      gameMode = GAME_MODE_HIT_OR_STAND;
+      var playerHandmessage = createPlayerHandmessage();
+      currentPlayer += 1;
+      console.log(currentPlayer);
+
+      return (
+        playerHandmessage +
+        "<br><br> Player " +
+        playerProfiles[currentPlayer - 1].id +
+        ", your hand is over 21, busted!<br> Player " +
+        playerProfiles[currentPlayer].id +
+        ", hit or stand?"
+      );
+    }
+  } else if (validation == "" && gameMode == GAME_MODE_HIT_OR_STAND) {
+    gameMode = GAME_MODE_HIT_OR_STAND;
+    return (
+      playerHandmessage +
+      "<br><br> Invalid. <br><br> Player " +
+      playerProfiles[currentPlayer].id +
+      ", it's still your turn, continue to hit or stand?<br><br> Enter Hit to draw another card or enter Stand to continue game."
+    );
+  } else if (Number(validation)) {
+    gameMode = GAME_MODE_HIT_OR_STAND;
+    return (
+      playerHandmessage +
+      "<br><br> Invalid. You entered a number. <br><br> Player " +
+      playerProfiles[currentPlayer].id +
+      ", it's still your turn, continue to hit or stand?<br><br> Enter Hit to draw another card or enter Stand to continue game."
+    );
+  } else if (
+    isNaN(validation) &&
+    !(validation == "HIT") &&
+    !(validation == "STAND")
+  ) {
+    gameMode = GAME_MODE_HIT_OR_STAND;
+    return (
+      playerHandmessage +
+      "<br><br> Invalid. Nan.<br><br> Player " +
+      playerProfiles[currentPlayer].id +
+      ", it's still your turn, continue to hit or stand?<br><br> Enter Hit to draw another card or enter Stand to continue game."
+    );
+  } else gameMode = GAME_MODE_DEALER_TURN;
+  var playerHandmessage = createPlayerHandmessage();
+  gameMode = GAME_MODE_DEALER_TURN;
+  console.log(gameMode);
+  return (
+    playerHandmessage +
+    "<br><br> Player " +
+    playerProfiles[currentPlayer].id +
+    ", your hand is over 21, busted!<br><br> Continue to dealer's turn"
+  );
+}
+function drawingDealerCard(deck) {
+  while (dealerProfile.dealerHandValue < 17) {
+    let dealerCard = deck.pop();
```

Comment:
You seem to use the deck.pop() and getValue and checkAce function together all the time. This is a hint to you that you could refactor these 3 things into a function on its own